addon-api/popup/Popup.js

```diff
@@ -14,4 +14,21 @@ export default class Popup {
   get isLightMode() {
     return scratchAddons.isLightMode;
   }
+
+  /**
+   * Gets the URL of the Scratch page that is selected, or null.
+   * @returns {Promise<?string>} - the URL
+   */
+  getSelectedTabUrl() {
+    return new Promise((resolve) => {
+      chrome.tabs.query(
+        {
+          active: true,
+          currentWindow: true,
+          url: "https://scratch.mit.edu/*",
```

Comment:
Don't forget in older Firefox versions we do support, we cannot query tabs by URL because we don't have access to the URL. We might want to do an ugly workaround that involves content scripts to make it work there.